tests/components/airly/test_init.py

```diff
@@ -110,15 +147,25 @@ async def test_update_interval(hass, aioclient_mock):
     aioclient_mock.get(
         "https://airapi.airly.eu/v2/measurements/point?lat=66.660000&lng=111.110000",
         text=load_fixture("airly_valid_station.json"),
+        headers=HEADERS,
     )
     entry.add_to_hass(hass)
     await hass.config_entries.async_setup(entry.entry_id)
     await hass.async_block_till_done()
+    instances = 2
 
+    assert aioclient_mock.call_count == 3
     assert len(hass.config_entries.async_entries(DOMAIN)) == 2
     assert entry.state == ENTRY_STATE_LOADED
-    for instance in hass.data[DOMAIN].values():
-        assert instance.update_interval == timedelta(minutes=30)
+
+    update_interval = set_update_interval(instances, REMAINING_RQUESTS)
+    future = utcnow() + update_interval
+    with patch("homeassistant.util.dt.utcnow", return_value=future):
```

Comment:
If we store the patch context manager mocks in variables we don't need to create new contexts when changing the return value of the mocks. We can just set a new return value on the mocks.

```py
with patch(...) as mock:
    mock.return_value = value_1
    ...
    mock return_value = value_2
```

Reply:
Fixed